src/diff/patch.js

```diff
@@ -205,4 +205,17 @@ function patchDOMElement(dom, newVNode, internal, globalContext, commitQueue) {
 			dom.firstChild
 		);
 	}
+
+	if ('value' in newProps && dom._isControlled) {
```

Comment:
After diffing the children we check whether the value got out of sync, if it did we update it. We also update the `_prevValue` to prepare for the next event hitting our controlled component